ui/hooks/useTransactionFunctionType.js

```diff
@@ -0,0 +1,42 @@
+import { useSelector } from 'react-redux';
+
+import { TransactionType } from '../../shared/constants/transaction';
+import { getKnownMethodData } from '../selectors';
+import { getNativeCurrency } from '../ducks/metamask/metamask';
+import { getTransactionTypeTitle } from '../helpers/utils/transactions.util';
+import { getMethodName } from '../helpers/utils/metrics';
+
+import { useI18nContext } from './useI18nContext';
+
+export const useTransactionFunctionType = (txData) => {
+  const t = useI18nContext();
+  const nativeCurrency = useSelector(getNativeCurrency);
+  const { txParams } = txData;
+  const methodData = useSelector(
+    (state) => getKnownMethodData(state, txParams.data) || {},
+  );
+
+  const isTokenApproval =
+    txData.type === TransactionType.tokenMethodSetApprovalForAll ||
+    txData.type === TransactionType.tokenMethodApprove;
+
+  const isContractInteraction =
+    txData.type === TransactionType.contractInteraction;
+
+  const isContractInteractionFromDapp =
+    (isTokenApproval || isContractInteraction) && txData.origin !== 'metamask';
```

Comment:
[nit] we can use const here: 'metamask' -> ORIGIN_METAMASK

Reply:
👍 done